### PR TITLE
Support for annotations

### PR DESCRIPTION
**Pull request by ahmedwaleedmalik**
**What this PR does / why we need it**:
This PR brings in support for configuring annotations for clusters, machine deployments, and worker nodes. Since we can't simply show all annotations without any filters, we have added admin settings to mark annotations as either hidden or protected. Protected annotations are displayed on the UI but users can't edit, delete, or add them by themselves.

**Which issue(s) this PR fixes**:

Fixes https://github.com/kubermatic/kubermatic/issues/11569

**What type of PR is this?**

/kind feature

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
* Dashboard now supports managing annotations for clusters, machine deployments, and nodes.
* Admin settings have been introduced for annotations. Admins can hide annotations or mark them as protected/read-only.
```

**Documentation**:

```documentation
NONE
```

### CODE REVIEW

**Comment by ahmedwaleedmalik** [comment]
I'm unable to run the integration tests locally. I documented the error at https://github.com/kubermatic/dashboard/issues/6828. @ahmadhamzh @Waseem826, can anyone of you check what's wrong with the SSH keys tests?

**Review by Waseem826 — COMMENTED** [review]
Couple of comments. PTAL

File: modules/web/src/app/cluster/details/cluster/component.ts [hunk 1]
```diff
@@ -368,6 +369,10 @@ export class ClusterDetailsComponent implements OnInit, OnDestroy {
       .subscribe(_ => this.goBack());
   }
 
+  getFilteredAnnotations(annotations: Record<string, string>): Record<string, string> {
+    return getVisibleAnnotations(annotations, this.adminSettings);
+  }
```
[the remark this answers is not in the document]
Comment (ahmedwaleedmalik) [reply modules/web/src/app/cluster/details/cluster/component.ts:374]: I think this should still be fine

File: modules/web/src/app/shared/components/annotation-form/component.ts [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+  isAnnotationChanged(index: number): boolean {
+    const annotation = this.annotationsArray.at(index);
+    const key = annotation.get('key').value;
+    if (key) {
+      return this.annotations[key] !== annotation.get('value').value && this._dialogModeService.isEditDialog;
+    }
+    return false;
+  }
+
+  private async initializeAnnotationSets(): Promise<void> {
```
[the remark this answers is not in the document]
Comment (ahmedwaleedmalik) [reply modules/web/src/app/shared/components/annotation-form/component.ts:183]: Updated

**Review by Waseem826 — APPROVED** [review]
/approve

**Pull request merged**